core/src/main/java/org/elasticsearch/common/path/PathTrie.java

```diff
@@ -218,6 +254,10 @@ public T retrieve(String path, Map<String, String> params) {
         if (strings.length > 0 && strings[0].isEmpty()) {
             index = 1;
         }
-        return root.retrieve(strings, index, params);
+        if (ignoreWildcards == true) {
```

Comment:
We can just write this as `if (ignoreWildcards)`; the `if (b == false)` thing is to avoid writing `if (!b)` because sometimes the `!` gets lost when reading, I guess.